File: src/services/instructions/loadstore/LD_B_REGX_MREGY.java
package emulator.source.loadstore

import emulator.engine.CpuContext
 import Instruction from '../Instruction'import REGISTER_VALUE_NAME_MAPPER from '@/constants/registers'

export default class LD_B_REGX_MREGY extends Instruction {
	public LD_B_REGX_MREGY(memory, address, source, 
 destination) {
		super(memory, address, source, destination)
		super.setAssembler("ld.b " + this.sdest + ", [" + this.ssource + "]")
	}

	
	exec ({ context, memory }) {
		int fixedAddr = fix(context.getReg(this.source) )
		if ((fixedAddr & 1) == 0)
			context.getReg(this.destination)  = (short)(context.memory[fixedAddr / 2] >> 8) & 0xFF
		else
			context.getReg(this.destination)  = (short)(context.memory[fixedAddr / 2] & 255) & 0xFF
		context.pc  += 2
	}
}

File: src/services/instructions/loadstore/LdBRegXMRegY.js

 import Instruction from '../Instruction'
import { REGISTER_VALUE_NAME_MAPPER } from '@/constants/registers'

const ASSEMBLER_INSTRUCTION_EXPRESSION = (sdestination, ssource) => `ld.b ${sdestination}, [${ssource}]`

export default class LdBRegXMRegY extends Instruction {
	constructor (memory, address, source, destination, symbolTable) {
		super(memory, address, source, destination, symbolTable)
		super.setAssembler(ASSEMBLER_INSTRUCTION_EXPRESSION(this.sdestination, this.ssource))
	}

	exec ({ context, memory }) {
		const fixedAddress = Instruction.fix(context[REGISTER_VALUE_NAME_MAPPER[this.source]])
		if ((fixedAddress & 1) == 0) {
			// TODO - PROBLEM : Moguc problem prilikom izvrsavanja, provjeriti
			// TODO: ->> (short)(context.memory[fixedAddr / 2] >> 8) & 0xFF
			context[REGISTER_VALUE_NAME_MAPPER[this.destination]]  = (memory[Math.floor(fixedAddress / 2)] >> 8) & 0xFF
		} else {
			// TODO: ->> (short)(context.memory[fixedAddr / 2] & 255) & 0xFF
			context[REGISTER_VALUE_NAME_MAPPER[this.destination]]  = (memory[Math.floor(fixedAddress / 2)] & 255) & 0xFF
		}

		context.pc  += 2
	}
}
